pkg/volume/configmap/configmap.go

```diff
@@ -216,16 +216,28 @@ func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap, defaultMode
 			fileProjection.Mode = *defaultMode
 			payload[name] = fileProjection
 		}
+		for name, data := range configMap.BinaryData {
+			fileProjection.Data = data
+			fileProjection.Mode = *defaultMode
+			payload[name] = fileProjection
+		}
 	} else {
 		for _, ktp := range mappings {
-			content, ok := configMap.Data[ktp.Key]
-			if !ok {
+			contentData, okData := configMap.Data[ktp.Key]
```

Comment:
This would be cleaner as:

```
var data []byte
if stringData, ok := configMap.Data[ktp.Key]; ok {
  data = []byte(stringData)
} else {
  data, ok = configMap.BinaryData[ktp.Key]
  if !ok {
    // return err
  }
}
```

Having the if block below is harder to read.